File: typescript-client-sdk/src/sdk/models/operations/getplansplanid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetPlansPlanIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=plan_id" })
  planId: string;
}


export class GetPlansPlanIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetPlansPlanIdPathParams;
}


export class GetPlansPlanIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  plan?: shared.Plan;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getplansplanid.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetPlansPlanIDPathParams struct {
	PlanID string `pathParam:"style=simple,explode=false,name=plan_id"`
}

type GetPlansPlanIDRequest struct {
	PathParams GetPlansPlanIDPathParams
}

type GetPlansPlanIDResponse struct {
	ContentType string
	Plan        *shared.Plan
	StatusCode  int
}
